pontoon/base/utils.py

```diff
@@ -398,17 +386,75 @@ def get_download_content(slug, code, part):
             vcs_translation.update_from_db(entity)
 
     resource_file.save(locale)
+    return locale_path, source_path
+
+
+def get_download_content(slug, code, part):
+    """
+    Get content of the file to be downloaded.
+
+    :param str slug: Project slug.
+    :param str code: Locale code.
+    :param str part: Resource path or Subpage name.
+    """
+    from pontoon.base.models import (
+        Locale,
+        Project,
+        Resource,
+    )
+
+    project = get_object_or_404(Project, slug=slug)
+    locale = get_object_or_404(Locale, code__iexact=code)
+
+    # Download a ZIP of all files if project has less than 10 resources
+    if 1 < project.resources.count() < 10:
+        files = {}
+
+        for resource in project.resources.all():
+            locale_path, source_path = _generate_file_for_download(resource, locale)
+
+            if not locale_path:
+                return None, None
+
+            files[resource.path] = (locale_path, source_path)
+
+        s = StringIO.StringIO()
+        zf = zipfile.ZipFile(s, "w")
+
+        for fpath, value in files.iteritems():
+            locale_path = value[0]
+            source_path = value[1]
+            _, fname = os.path.split(fpath)
+            zf.write(locale_path, fname)
+
+            os.remove(locale_path)
+            if source_path:
+                os.remove(source_path)
+
+        zf.close()
+
+        content = s.getvalue()
+        filename = project.slug + '.zip'
```

Comment:
Usually localizers download these files for personal use and they usually only localize into one locale. We also don't put locale names when a single file is downloaded.